collector/mssql.go

```diff
@@ -19,18 +19,53 @@
 package collector
 
 import (
+	"errors"
 	"fmt"
+	"regexp"
+	"sync"
+	"time"
 
 	"github.com/StackExchange/wmi"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/log"
 	"golang.org/x/sys/windows/registry"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
 )
 
-type sqlInstancesType map[string]string
+var (
+	mssqlCollectorWhitelist = kingpin.Flag(
+		"collector.mssql.class-whitelist",
+		"Regexp of mssql WMI classes to whitelist. Name must both match whitelist and not match blacklist to be included.",
+	).Default(".+").String()
+
+	mssqlCollectorBlacklist = kingpin.Flag(
+		"collector.mssql.class-blacklist",
+		"Regexp of mssql WMI classes to blacklist. Name must both match whitelist and not match blacklist to be included.",
+	).Default("").String()
+
+	mssqlPrintCollectors = kingpin.Flag(
+		"collectors.mssql.class-print",
+		"If true, print available mssql WMI classes",
+	).Bool()
+
+	mssqlScrapeDurationDesc = prometheus.NewDesc(
```

Comment:
Not sure this really works? We'll end up with two identically named metrics, which I'm fairly sure results in an error?
Any reason not to throw it in a dedicated metric `wmi_mssql_subcollector_duration_seconds` (or similar) instead?

Reply:
it actually does work.  the resulting metrics look like this: 
```
# HELP wmi_exporter_collector_duration_seconds wmi_exporter: Duration of a collection.
# TYPE wmi_exporter_collector_duration_seconds gauge
wmi_exporter_collector_duration_seconds{collector="cpu"} 1.5950872999999999
wmi_exporter_collector_duration_seconds{collector="cs"} 1.5630859
wmi_exporter_collector_duration_seconds{collector="logical_disk"} 0.9350555
wmi_exporter_collector_duration_seconds{collector="mssql"} 2.6281417
wmi_exporter_collector_duration_seconds{collector="mssql_availreplica"} 1.6430905
wmi_exporter_collector_duration_seconds{collector="mssql_bufman"} 1.6100894000000001
wmi_exporter_collector_duration_seconds{collector="mssql_databases"} 1.9571038
wmi_exporter_collector_duration_seconds{collector="mssql_dbreplica"} 2.6281417
wmi_exporter_collector_duration_seconds{collector="mssql_genstats"} 2.4711424
wmi_exporter_collector_duration_seconds{collector="mssql_locks"} 2.5011408
wmi_exporter_collector_duration_seconds{collector="mssql_memmgr"} 1.970111
wmi_exporter_collector_duration_seconds{collector="mssql_sqlstats"} 1.6220949
wmi_exporter_collector_duration_seconds{collector="net"} 1.5520828
wmi_exporter_collector_duration_seconds{collector="os"} 1.5080826
wmi_exporter_collector_duration_seconds{collector="process"} 2.6321453
wmi_exporter_collector_duration_seconds{collector="service"} 2.4591326000000002
wmi_exporter_collector_duration_seconds{collector="system"} 2.5171416
wmi_exporter_collector_duration_seconds{collector="tcp"} 1.5230799
```

Reply:
Interestnig. But I think that would be best to change. It might either be that the client library is clever and helpful, or it might even be some luck. Either way, I don't think it is a good idea to have multiple "different" metrics that end up with the same name.